### DISCUSSION TITLE
Enabling Debug Logging in OTBR for Better Insights

### DISCUSSION THREAD

**Discussion by cedriczou13** [openthread/openthread#9469] [General]
Hi everyone,

I'm trying to get a deeper understanding of OTBR's workings. To do this, I'd like to enable a debug logging feature during its build configuration. Can anyone provide guidance or best practices on how to integrate this debug logging feature effectively?

**Comment by Alami-Amine** [comment]
Do you mean an OTBR built with this implementation: https://github.com/openthread/ot-br-posix ?
To get the most possible logs add a switch -d 7 when running the OTBR.

and for Build Options you can use:  OT_FULL_LOGS=ON and OT_LOG_LEVEL=DEBG

**Reply by cedriczou13** [reply]
@aalami90 Thank you for clarifying. I was indeed referring to ot-br-posix. 
Could you guide me on how to add the '-d 7' switch when executing OTBR?

**Reply by Alami-Amine** [reply]
you can add it to the beginning of OTBR_AGENT_OPTS in the otbr-configuration file --> check step 4 here [https://openthread.io/guides/border-router/build](url)

**Comment by abtink** [comment]
There are different log-levels in OT:
- `DEBG` is useful but can be very verbose (every sent/received frame is fully dumped) - It is often used when debugging specific issue (specially if we suspect any issue with radio platform or other platform drivers)
- `INFO` is a good log level to use for seeing overall behavior of the device (I would suggest this to get insight about OT stack)
- There are other log level which reduce the amount of logs. `NOTE` level provides less logs than `INFO` and includes the more important logs (state changes, etc).

**Comment by jwhui** [comment]
In case it helps, we have a guide on [OpenThread Logs](https://openthread.io/guides/build/logs).

**Comment by cedriczou13** [comment]
Thanks, Everyone !!!
